### PR TITLE
Add regex debugging mode description

### PR DESCRIPTION
**Pull request by DL6ER**
See related PR https://github.com/pi-hole/FTL/pull/326

### CODE REVIEW

File: docs/ftldns/regex/overview.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+A regular expression, or RegEx for short, is a pattern that can be used for building arbitrarily complex blocking rules in *FTL*DNS.
+We implement the ERE flavor similar to the one used by the UNIX `egrep` (or `grep -E`) command.
```
Comment (dschaper) [inline docs/ftldns/regex/overview.md]: Suggestion:

We implement the POSIX Extended Regular Expressions <Link if there is one> similar...

File: docs/ftldns/regex/overview.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+Our implementation is computationally inexpensive as each domain is only checked once for a match (if you query `google.com`, it will be checked against your RegEx. Any subsequent query to the same domain will not be checked again until you restart `pihole-FTL`).
```
Comment (dschaper) [inline docs/ftldns/regex/overview.md]: Our implementation is light and fast as each...

File: docs/ftldns/regex/overview.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+## How to use regular expressions for blocking
+*FTL*DNS reads in regular expression filters from `/etc/pihole/regex.list` (one expression per line, lines starting with `#` will be skipped).
+To tell *FTL*DNS to reload the list, either:
+
+- Execute the `>recompile-regex` API command (`echo ">recompile-regex" | nc localhost 4711`)
```
Comment (dschaper) [inline docs/ftldns/regex/overview.md]: 4711) or

File: docs/ftldns/regex/overview.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+- Send `SIGHUP` to `pihole-FTL` (`sudo killall -SIGHUP pihole-FTL`)
```
Comment (dschaper) [inline docs/ftldns/regex/overview.md]: FTL`) or

File: docs/ftldns/regex/overview.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+## Pi-hole Regex debugging mode
+To ease the usage of regular expression filters in *FTL*DNS, we offer a regex debugging mode. Set
+```
+REGEX_DEBUGMODE=true
+```
+in your `/etc/pihole/pihole-FTL.conf` and restart `pihole-FTL` to enable or disable this mode.
+
+Once the debugging mode is enabled, each match will be logged to `/var/log/pihole-FTL.log` in the following format:
+```
+[2018-07-17 17:40:51.304] DEBUG: Regex in line 2 ("((^)|(\.))twitter\.") matches "whatever.twitter.com"
```
Comment (Mcat12) [inline docs/ftldns/regex/overview.md]: The format of this line might change before the FTL PR is merged.

**Pull request merged**